aspnet/mvc/views/forms/sample/final/Views/Demo/EditColor.cshtml

```diff
@@ -0,0 +1,17 @@
+﻿@model Person
+@{
+    var index = (int)ViewData["index"];
+}
+
+<form asp-controller="ToDo" asp-action="Edit" method="post">
+    @*
+        Move the following two lines to a template to avoid
+        evaluating index twice.
+        <label asp-for="Colors[index]" />
```

Comment:
@dougbu  I'm going to add a :note: to the doc
Never use self-closing HTML elements with tag helpers ( with the exception of  `<input/>`  that works with a self closing tag).